compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4374,6 +4377,64 @@ private STNode parseObjectTypeDescriptor() {
                 objectMembers, closeBrace);
     }
 
+    /**
+     * Parse object constructor expression.
+     *
+     * object-constructor-expr :=
+     *    [annots] [client] object [type-reference] {
+     *       object-member*
+     *    }
+     * object-member := object-field | method-defn
+     *
+     * @param isRhsExpr
+     * @param annots
+     * @return Parsed node
+     */
+    private STNode parseObjectConstructorExpressionNode(boolean isRhsExpr, STNode annots) {
+
+        startContext(ParserRuleContext.OBJECT_CONSTRUCTOR);
+
+        STNode objectTypeQualifier;
+        STToken nextToken = peek();
+        if (nextToken.kind == SyntaxKind.CLIENT_KEYWORD) {
+            startContext(ParserRuleContext.CLIENT_KEYWORD);
+            objectTypeQualifier = parseClientKeyword();
```

Comment:
Since we come after checking next token is `CLIENT_KEYWORD`, we can directly use `consume()` without going `parseClientKeyword()`